### PR TITLE
Strip of users home path from share api message

### PR DESCRIPTION
**Pull request by nickvergessen**
* Share a folder without reshare permissions
* Try to share that file e.g. in talk (https://github.com/nextcloud/spreed/pull/2976)

### Before
You are not allowed to share /445d1288-a75a-44df-87fd-bb4c72bdc488/files/Resharing disabled/1P8A0501.jpg

### After
You are not allowed to share /Resharing disabled/1P8A0501.jpg

### CODE REVIEW

**Review by blizzz — APPROVED** [review]

**Comment by blizzz** [comment]
tests needs adjustment tho 

```
1) Test\Share20\ManagerTest::testGeneralChecks with data set #41 (Mock_IShare_ff82758b Object (...), 'Can’t increase permissions of ', true) 
TypeError: Argument 2 passed to OC\Share20\Manager::getVisiblePath() must be of the type string, null given, called in /drone/src/lib/private/Share20/Manager.php on line 336 

/drone/src/lib/private/Share20/Manager.php:362 
/drone/src/lib/private/Share20/Manager.php:336 
/drone/src/tests/lib/TestCase.php:210 
/drone/src/tests/lib/Share20/ManagerTest.php:715 
```

File: lib/private/Share20/Manager.php [hunk 1]
```diff
@@ -289,7 +289,7 @@ protected function generalCreateChecks(\OCP\Share\IShare $share) {
 
 		// Check if we actually have share permissions
 		if (!$share->getNode()->isShareable()) {
-			$message_t = $this->l->t('You are not allowed to share %s', [$share->getNode()->getPath()]);
+			$message_t = $this->l->t('You are not allowed to share %s', [$this->getVisiblePath($userFolderPath, $share->getNode()->getPath())]);
```
Comment (icewind1991) [inline lib/private/Share20/Manager.php]: `$userFolder->getRelativePath($node->getPath())` should do what you want

Comment (nickvergessen) [reply lib/private/Share20/Manager.php]: NIce, thanks

**Comment by nickvergessen** [comment]
Fixed tests

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**